cmake/CheckHeaders.cmake

```diff
@@ -84,6 +84,11 @@ function (seastar_check_self_contained library)
   list (REMOVE_DUPLICATES includes)
   target_include_directories (${check_lib}
     PRIVATE ${includes})
+  get_target_property (system_includes ${library} INTERFACE_SYSTEM_INCLUDE_DIRECTORIES)
+  if (system_includes)
+    target_include_directories (${check_lib} SYSTEM
+      PRIVATE ${system_includes})
```

Comment:
Maybe we can disable those warnings with #pragma instead, to avoid silencing all warnings.

Reply:
yeah, even better.